src/dynamic_boundary_conditions/river/river_data_from_niwa.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+"""
+Fetch REC data in New Zealand from NIWA using the ArcGIS REST API.
+"""
+
+import logging
+from typing import Tuple, List, Dict, Union
+import asyncio
+
+import aiohttp
+import requests
+import geopandas as gpd
+import pandas as pd
+from shapely.geometry import LineString
+from sqlalchemy.engine import Engine
+
+from src.digitaltwin.utils import get_nz_boundary
+
+log = logging.getLogger(__name__)
+
+# URL for retrieving REC data from NIWA using the ArcGIS REST API
+REC_API_URL = "https://gis.niwa.co.nz/server/rest/services/HYDRO/Flood_Statistics_Henderson_Collins_V2/MapServer/2"
+
+
+def get_feature_layer_record_counts(url: str = REC_API_URL) -> Tuple[int, int]:
+    """
+    Retrieves the maximum and total record counts from the REC feature layer.
+
+    Parameters
+    ----------
+    url : str = REC_API_URL
+        The URL of the REC feature layer. Defaults to 'REC_API_URL'.
```

Comment:
```suggestion
        The URL of the REC feature layer. Defaults to `REC_API_URL`.
```
I think the single quote for a variable should be avoided so that people don't mistake it for a `str`

Reply:
changed :)